src/whitelist.hpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (C) 2022, KNS Group LLC (YADRO).
+ */
+#pragma once
+
+#include <filesystem>
+#include <set>
+
+namespace fs = std::filesystem;
+
+namespace fssync
+{
+
+namespace details
+{
+struct PathsComparer
+{
+    /**
+     * @brief Compare the begining of strings.
+     *
+     * For example:
+     * The comparison of the strings like `etc/systemd` and
```

Comment:
No first `The` please. The second `the` is optional as `strings` is plural.

Reply:
fixed